### ISSUE TITLE
cannot display some unicode characters

### ISSUE THREAD

**Issue by zaoqi** [koreader/koreader#5277]
* KOReader version: KOReader 2019.08 "Starling"
* Device: KPW4

#### Issue

Some unicode symbols are shown as `?`:

```
≡
⟨
⟩
∎
```
But they are displayed properly in dictionary lookup.

#### Steps to reproduce

1. Download https://agda-zh.github.io/PLFA-zh/
```
wget -rpk https://agda-zh.github.io/PLFA-zh/
```

2. open agda-zh.github.io/PLFA-zh/Induction/index.html

**Comment by NiLuJe** [comment]
We're tributary to whatever the coverage of our fonts is, and then whatever is set as the current & fallback font.

Given the specific examples, you'd probably want to install & set as default or fallback a maths-oriented font.

**Comment by NiLuJe** [comment]
Or switch your fallback to Free Serif, which does cover these glyphs.

Or build a franken-fallback manually, like what @poire-z showed how to do recently w/ a fontforge script.

**Comment by zaoqi** [comment]
@NiLuJe
But they are displayed properly in dictionary lookup.

**Comment by Frenzie** [comment]
Pinging @poire-z.

**Comment by zaoqi** [comment]
@NiLuJe How to switch fallback/install/set as default a font? `/mnt/us/fonts` ?

**Comment by Frenzie** [comment]
@zaoqi You can switch fonts in the document → fonts menu, and long press to set default/fallback.

**Comment by zaoqi** [comment]
@NiLuJe 
It works:
fallback = FreeSans
default = Noto Sans CJK SC

**Comment by Frenzie** [comment]
That confused me, but for our GUI code we basically have a fallback consisting of all possible fonts:

https://github.com/koreader/koreader/blob/dd4bfb0a454aa07a3b66e424b6ef5db87dd2d7b8/frontend/ui/font.lua#L77-L82

**Comment by poire-z** [comment]
We discussed our fonts' coverage a bit in #3942, and the order you show just above must be what's best.

But there are still some area not covered by any font, for example some symbols (which got better with the recent font bump, but still some missing):
<kbd>![image](https://user-images.githubusercontent.com/24273478/63878321-a972e700-c9c9-11e9-92c4-3cf7d36d4459.png)</kbd>

As for crengine, there are indeed only 2 fonts used (if no other embedded font in the EPUB), as described at https://github.com/koreader/crengine/issues/307#issuecomment-523972046.
(May be we could hardcode 3 or 4 of our fallback fonts from above in crengine, that would be checked after the user default font and the user fallback font.)

Playing lately with other scripts (https://r12a.github.io/scripts/phrases), there are some missing coverage for some of these, but worse: some glyphs/features are wrong for some fonts (don't remember which), and harfbuzz renders some of ours ugly (while the text is fine looking with some other font).
Note that we have currently some mess with harfbuzz ("best") use of the fallback font (some bad spacing/advance...)

And my franken-fallback font actually can show the worse of all with some text :| (but I haven't updated if for some years, might be better after redoing it from our recent fonts bump).
It would be nice to have some good, complete and correct fonts, if we are gonna bring RTL support in crengine (and possibly in frontend? might not be that complicated now that I got a grasp on fribidi and harfbuzz, some small ffi wrappers could be enough).

(I still have to fix a few font drawing bugs to be certain of what I said just above about our fonts thus :)
